climada/hazard/tc_tracks_synth.py

```diff
@@ -166,6 +206,170 @@ def _one_rnd_walk(track, ens_size, ens_amp0, ens_amp, max_angle, rnd_vec):
     return ens_track
 
 
+def _random_uniform_ac(n_ts, autocorr, time_step_h):
+    """Generate a series of autocorrelated uniformly distributed random numbers.
+
+    This implements the algorithm described here to derive a uniformly distributed
+    series with specified autocorrelation (here at a lag of 1 hour):
+    https://stats.stackexchange.com/questions/48086/
+        algorithm-to-produce-autocorrelated-uniformly-distributed-number
+    Autocorrelation is specified at a lag of 1 hour. To get a time series at a
+    different temporal resolution (time_step_h), an hourly time series is generated
+    and resampled (using linear interpolation) to the target resolution.
+
+    Parameters:
+        n_ts (int): length of the series
+        autocorr: autocorrelation (between -1 and 1)
+
+    Returns:
+        numpy.ndarray of length n
+    """
+    # generate autocorrelated 1-hourly perturbations, so first create hourly
+    #   time series of perturbations
+    n_ts_hourly_exact = n_ts * time_step_h
+    n_ts_hourly = int(np.ceil(n_ts_hourly_exact))
+    x = np.random.normal(size=n_ts_hourly)
+    theta = np.arccos(autocorr)
+    for i in range(1, len(x)):
+        x[i] = _h_ac(x[i - 1], x[i], theta)
+    # scale x to have magnitude [0,1]
+    x = (x + np.sqrt(3)) / (2 * np.sqrt(3))
+    # resample at target time step
+    x_ts = np.interp(np.arange(start=0, stop=n_ts_hourly_exact, step=time_step_h),
+                     np.arange(n_ts_hourly), x)
+    return x_ts
+
+
+@jit
+def _h_ac(x, y, theta):
+    # https://stats.stackexchange.com/questions/48086/
+    #   algorithm-to-produce-autocorrelated-uniformly-distributed-number
+    # Note that the definition of Gamma is not very clear there, but the
+    # formulation below does what the text says.
+    # Tests indicated that this works, while the following fails:
+    # gamma = np.abs(np.mod(np.mod(theta, np.pi), np.pi / 2) - np.pi / 4)
+    gamma = np.abs(np.mod(theta, np.pi) - \
+                   np.floor((np.mod(theta, np.pi) / (np.pi / 2)) + 0.5) * np.pi / 2)
+    # np.abs(dat['theta2'] - np.floor((dat['theta2'] / (np.pi / 2)) + 0.5) * np.pi / 2)
+    return 2 * np.sqrt(3) * (_f_ac(np.cos(theta) * x + np.sin(theta) * y, gamma) - 1 / 2)
+
+
+@jit
+def _f_ac(z, theta):
+    """F transform for autocorrelated random uniform series generation
+
+    This is function F defined here:
+    https://stats.stackexchange.com/questions/48086/algorithm-to-produce-autocorrelated-uniformly-distributed-number
+    The CDF of Y.
+
+    Parameters:
+        z (float): longitude coordinates, in decimal degrees
+        theta (float): arccos of the autocorrelation
+
+    Returns:
+        float (CDF at z)
+    """
```

Comment:
According to CLIMADA's coding conventions, docstrings should follow the NumPy style. From one of the developer meetings, I remember that, while older code in CLIMADA is still allowed to contain google-style docstrings, new code is required to follow the new convention. Old docstrings should be converted to the new convention whenever they are touched.

Reply:
yes, e837bc88958cd756867fef5a7ef8d79571c8981c